File: number-of-1-bits/river20s.py
class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        Time complexity: O(1)
        Space complexity: O(1)
        """
        count = 0
        for _ in range(32):
            count += n & 1
            n >>= 1
        return count

File: reverse-linked-list/river20s.cpp
class Solution {
    /*
    * Time Complexity: O(n)
    * Space Complexity: O(1)
    */
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* prev = nullptr;
        ListNode* current = head;
        ListNode* next_temp = nullptr;

        while (current != nullptr) {
            next_temp = current->next;
            current->next = prev;
            prev = current;
            current = next_temp;
        } 
        return prev;       
    }
};
